flaskapp/view_functions/ivrflow.py

```diff
@@ -371,3 +371,10 @@ def get_profile():
     patient["type"] = pat.type
     print(patient)
     return jsonify(patient)
+
+# http://127.0.0.1:5000/new_user?username=testuser&&type=patient&&timezone=US/Pacific&&calltime=5:30:00&&phone=123-456-789
+def new_user():
+    all_args = request.args.to_dict()
+    rec1 = Patient.create(**all_args)
```

Comment:
If this API endpoint will be publicly available, somebody can form requests with unexpected GET parameters 
#e.g http://127.0.0.1:5000/new_user?username=testuser&&type=patient&&timezone=US/Pacific&&calltime=5:30:00&&phone=123-456-789&bubble=boom
and this crashes the program; Another security issue: What if someone performs a lot of similar requests to pollute the database with fake patients...

Reply:
Totally legitimate concerns and great points. So would you suggest I create a POST route instead which will hide new user details in request body? Thanks for bringing up these good points. Alex suggested I maybe not use a POST route so maybe this is more of a first step towards using Postgres only as opposed to a full working solution. Let me know your thoughts and I will change accordingly thanks a lot

Reply:
maybe I could do something like this like you did with get_profile() using request.json?

```
def new_user():
    content = request.json
    rec1 = Patient.create(**content)
    rec1.save()
    return {"success" : 200, "newuser" : model_to_dict(rec1)} 

import requests
res = requests.post('http://localhost:5000/add_user', json= 
   {"username":"newuser1"})
if res.ok:
    print(res.json())
```

Reply:
I didn't commit to the main branch yet ("get_profile" is not my work.. ); From here, I see the following issues with any of the API currently implemented: 
1) we didn't apply any validation to received data (it is dangerous to pass data directly to the database, we need to use something like pydantic, marshmallow); 
2) we need to use some authentication (e.g. via JWT) when hitting API endpoints. I changed base orm-models (all changes I made are in dev branch now) and hope I will complete the testing environment soon. When I do it, It would be possible to merge dev and main branches. It would be great if we were able to test all functionality automatically.

Reply:
Theoretically, if the "new_user" function works as expected, we can merge it; but we definitely need to handle validation and security issues in the nearest feature.

Reply:
@scidam these are really great suggestions, thanks a lot for taking some time to share your thoughts on the best way to move forward. I totally agree that it would be awesome to validate the data submitted as well as some kind of token to authenticate like you mentioned. Not an expert in those areas but I will do some reading about them.